src/services/user.py

```
@@ -0,0 +1,784 @@
+            user.major_id = body.major_id
+        if body.status:
+            user.status = body.status
+        if body.discord_id:
+            user.discord_id = body.discord_id
+        if body.discord_name:
+            user.discord_name = body.discord_name
+
+        try:
+            self.user_repository.update(user)
+        except IntegrityError:
+            raise HTTPException(409, detail="unique field already exists")
```

Ruff (0.14.6)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)